LocalReferenceLinkRel/explainer.md

```diff
@@ -0,0 +1,105 @@
+content location of future work and discussions.
+
+* This document status: **Active**
+* Expected venue: [WHAT Working Group](https://whatwg.org/)
+* Current version: this document
+
+## Introduction
+Modern web development practices have converged towards building reusable components instead of building monolithic documents. Frameworks such as React and Angular have popularized this approach, which has made its way natively to the web platform via Web Components. 
+
+Web Components allow for defining custom elements, and these custom elements often rely on Shadow DOM for ID encapsulation. But Shadow DOM doesn't just encapsulate ID's - it encapsulates style as well. The current preferred approach
+for styling elements under a Shadow DOM is through `<link>` tags, but these tags have several downsides, namely that they must be an external file or a dataURI. The performance and developer ergonomics of this
+situation are not ideal, and developers using Custom Elements have expressed strong interest in addressing these issues.
+
+This explainer proposes a solution to this situation by allowing another option for sharing styles via the `<link>` tag - local references to `<style>` objects.
+
+## Goals
+* Allow for developers to share styles defined in the Light DOM document into Shadow DOM trees.
+* Allow developers to define their CSS upfront in the same document as their HTML when using Shadow DOM.
+* Define a mechanism that will allow for more selective style sharing via CSS `@sheet`.
+
+## Non-goals
+* Anything specific to `@sheet` should be discussed in its dedicated proposal.
+
+## Proposal - Local References for Link Rel Tags
+
+We propose supporting fragment identifiers that reference the `id` attribute of a `<style>` or `<link>` tag:
+
+```html
+<style id="inline_styles">
+  p { color: blue; }
+</style>
+<p>Outside Shadow DOM</p>
+<template shadowrootmode="open">
+  <link rel="inline-stylesheet" href="#inline_styles">
+  <p>Inside Shadow DOM</p>
+</template>
+```
+With this functionality, the text "Inside Shadow DOM" will by styled blue, due to the `<link rel="inline-stylesheet" href="#inline_styles">` node's stylesheet applying (along with the `color: blue` rule applying via the `p` selector in that stylesheet).
+
+A new value for the `rel` attribute is used here to prevent legacy engines from performing an unnecessary fetch with this markup. The namne `inline-stylesheet` is preliminary and open to discussion.
+
+### Linking to another `<link>` tag
+
+```html
+<link rel="stylesheet" href="foo.css" id="inline_styles">
+<p>Outside Shadow DOM</p>
+<template shadowrootmode="open">
+  <link rel="inline-stylesheet" href="#inline_styles">
+  <p>Inside Shadow DOM</p>
+</template>
+```
+
+Developers may want to link to other `<link>` tags. These `<link>` references may be internal files (via `inline-stylesheet`), or external files. Loop detection may be necessary for references to another `inline-stylesheet`. 
```

Comment:
"Developers may want to link to other `<link>` tags." -> "Developers may want to link to styles pulled into another `<link>` tag."

(Feel free to ignore this if it doesn't quite capture this accurately)